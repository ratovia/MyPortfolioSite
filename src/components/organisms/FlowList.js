import React from "react"
import styled from "styled-components"
import { FrederickaH2 } from "../atoms/FrederickaText"
import { FlowItem } from "../molecules/FlowItem"

const FlowListArea = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  padding: 60px 30px;
`

const TitleText = styled.div`
  align-self: flex-end;
  margin-bottom: 60px;
`

const FlowItemsArea = styled.div`
  width: 100%;
`

export const FlowList = ({ flows }) => {
  return (
    <FlowListArea>
      <TitleText>
        <FrederickaH2 label="skills" color="white"></FrederickaH2>
      </TitleText>
      <FlowItemsArea>
        {flows.map(flow => {
          return (
            <FlowItem
              startDate={flow.startDate}
              endDate={flow.endDate}
              orgName={flow.orgName}
              period={flow.period}
              content={flow.content}
              batchs={flow.batchs}
              key={flow.orgName}
            ></FlowItem>
          )
        })}
      </FlowItemsArea>
    </FlowListArea>
  )
}
