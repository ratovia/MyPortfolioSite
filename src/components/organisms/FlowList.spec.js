import React from "react"
import renderer from "react-test-renderer"
import { FlowList } from "./FlowList"

describe("FlowList", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <FlowList
          flows={[
            {
              startDate: "2019年8月",
              endDate: "現在",
              orgName: "株式会社xxx",
              period: "1年間",
              content:
                "helloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworld",
              batchs: ["営業", "エンジニア", "セールス", "労務"],
            },
            {
              startDate: "2019年8月",
              endDate: "現在",
              orgName: "株式会社yyy",
              period: "1年間",
              content:
                "helloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworld",
              batchs: ["営業", "エンジニア", "セールス", "労務"],
            },
            {
              startDate: "2019年8月",
              endDate: "現在",
              orgName: "株式会社zzz",
              period: "1年間",
              content:
                "helloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworld",
              batchs: ["営業", "エンジニア", "セールス", "労務"],
            },
          ]}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
