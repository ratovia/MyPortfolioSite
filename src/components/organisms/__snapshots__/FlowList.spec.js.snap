// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowList renders correctly 1`] = `
<div
  className="sc-fotPbf btbdvj"
>
  <div
    className="sc-fFehDp khUShO"
  >
    <h2
      className="sc-dkPtyc FkeEH"
    >
      <p
        className="sc-bdvvaa bKtFxq"
        color="white"
      >
        skills
      </p>
    </h2>
  </div>
  <div
    className="sc-bkkfTU iSsTXg"
  >
    <div
      className="sc-iqsfdx cgIUbk"
    >
      <div
        className="sc-crHlIS fMvCKS"
      >
        <h6
          className="sc-iCfLBT ebzyty"
        >
          <p
            className="sc-eCImvq kBjSVF"
            color="#C7C7C7"
          >
            2019年8月
          </p>
        </h6>
        <h6
          className="sc-iCfLBT ebzyty"
        >
          <p
            className="sc-eCImvq kBjSVF"
            color="#C7C7C7"
          >
            〜
          </p>
        </h6>
        <h6
          className="sc-iCfLBT ebzyty"
        >
          <p
            className="sc-eCImvq kBjSVF"
            color="#C7C7C7"
          >
            現在
          </p>
        </h6>
      </div>
      <div
        className="sc-egiSv bgOump"
      >
        <div
          className="sc-pVTma gzkKVI"
        >
          <div
            className="sc-jrQzUz dVATvR"
          />
          <div
            className="sc-kDThTU bxhJKP"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              style={
                Object {
                  "fontSize": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="sc-bqiQRQ iqUMFD"
      >
        <div
          className="sc-ksdxAp hXqVet"
        >
          <h5
            className="sc-gKckTs dFXzsw"
          >
            <p
              className="sc-eCImvq eTrCnx"
              color="white"
            >
              株式会社xxx
            </p>
          </h5>
          <h6
            className="sc-iCfLBT ebzyty"
          >
            <p
              className="sc-eCImvq kBjSVF"
              color="#C7C7C7"
            >
              1年間
            </p>
          </h6>
        </div>
        <div
          className="sc-ksdxAp hXqVet"
        >
          <span
            className="sc-hBURRC hTBict"
          >
            <div
              className="sc-furvIG dJoCcd"
            >
              <p
                className="sc-eCImvq dVIdtT"
                color="#424242"
              >
                営業
              </p>
            </div>
          </span>
          <span
            className="sc-hBURRC hTBict"
          >
            <div
              className="sc-furvIG dJoCcd"
            >
              <p
                className="sc-eCImvq dVIdtT"
                color="#424242"
              >
                エンジニア
              </p>
            </div>
          </span>
          <span
            className="sc-hBURRC hTBict"
          >
            <div
              className="sc-furvIG dJoCcd"
            >
              <p
                className="sc-eCImvq dVIdtT"
                color="#424242"
              >
                セールス
              </p>
            </div>
          </span>
          <span
            className="sc-hBURRC hTBict"
          >
            <div
              className="sc-furvIG dJoCcd"
            >
              <p
                className="sc-eCImvq dVIdtT"
                color="#424242"
              >
                労務
              </p>
            </div>
          </span>
        </div>
        <div
          className="sc-ksdxAp hXqVet"
        >
          <h6
            className="sc-iCfLBT ebzyty"
          >
            <p
              className="sc-eCImvq eTrCnx"
              color="white"
            >
              helloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworld
            </p>
          </h6>
        </div>
      </div>
    </div>
    <div
      className="sc-iqsfdx cgIUbk"
    >
      <div
        className="sc-crHlIS fMvCKS"
      >
        <h6
          className="sc-iCfLBT ebzyty"
        >
          <p
            className="sc-eCImvq kBjSVF"
            color="#C7C7C7"
          >
            2019年8月
          </p>
        </h6>
        <h6
          className="sc-iCfLBT ebzyty"
        >
          <p
            className="sc-eCImvq kBjSVF"
            color="#C7C7C7"
          >
            〜
          </p>
        </h6>
        <h6
          className="sc-iCfLBT ebzyty"
        >
          <p
            className="sc-eCImvq kBjSVF"
            color="#C7C7C7"
          >
            現在
          </p>
        </h6>
      </div>
      <div
        className="sc-egiSv bgOump"
      >
        <div
          className="sc-pVTma gzkKVI"
        >
          <div
            className="sc-jrQzUz dVATvR"
          />
          <div
            className="sc-kDThTU bxhJKP"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              style={
                Object {
                  "fontSize": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="sc-bqiQRQ iqUMFD"
      >
        <div
          className="sc-ksdxAp hXqVet"
        >
          <h5
            className="sc-gKckTs dFXzsw"
          >
            <p
              className="sc-eCImvq eTrCnx"
              color="white"
            >
              株式会社yyy
            </p>
          </h5>
          <h6
            className="sc-iCfLBT ebzyty"
          >
            <p
              className="sc-eCImvq kBjSVF"
              color="#C7C7C7"
            >
              1年間
            </p>
          </h6>
        </div>
        <div
          className="sc-ksdxAp hXqVet"
        >
          <span
            className="sc-hBURRC hTBict"
          >
            <div
              className="sc-furvIG dJoCcd"
            >
              <p
                className="sc-eCImvq dVIdtT"
                color="#424242"
              >
                営業
              </p>
            </div>
          </span>
          <span
            className="sc-hBURRC hTBict"
          >
            <div
              className="sc-furvIG dJoCcd"
            >
              <p
                className="sc-eCImvq dVIdtT"
                color="#424242"
              >
                エンジニア
              </p>
            </div>
          </span>
          <span
            className="sc-hBURRC hTBict"
          >
            <div
              className="sc-furvIG dJoCcd"
            >
              <p
                className="sc-eCImvq dVIdtT"
                color="#424242"
              >
                セールス
              </p>
            </div>
          </span>
          <span
            className="sc-hBURRC hTBict"
          >
            <div
              className="sc-furvIG dJoCcd"
            >
              <p
                className="sc-eCImvq dVIdtT"
                color="#424242"
              >
                労務
              </p>
            </div>
          </span>
        </div>
        <div
          className="sc-ksdxAp hXqVet"
        >
          <h6
            className="sc-iCfLBT ebzyty"
          >
            <p
              className="sc-eCImvq eTrCnx"
              color="white"
            >
              helloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworld
            </p>
          </h6>
        </div>
      </div>
    </div>
    <div
      className="sc-iqsfdx cgIUbk"
    >
      <div
        className="sc-crHlIS fMvCKS"
      >
        <h6
          className="sc-iCfLBT ebzyty"
        >
          <p
            className="sc-eCImvq kBjSVF"
            color="#C7C7C7"
          >
            2019年8月
          </p>
        </h6>
        <h6
          className="sc-iCfLBT ebzyty"
        >
          <p
            className="sc-eCImvq kBjSVF"
            color="#C7C7C7"
          >
            〜
          </p>
        </h6>
        <h6
          className="sc-iCfLBT ebzyty"
        >
          <p
            className="sc-eCImvq kBjSVF"
            color="#C7C7C7"
          >
            現在
          </p>
        </h6>
      </div>
      <div
        className="sc-egiSv bgOump"
      >
        <div
          className="sc-pVTma gzkKVI"
        >
          <div
            className="sc-jrQzUz dVATvR"
          />
          <div
            className="sc-kDThTU bxhJKP"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              style={
                Object {
                  "fontSize": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="sc-bqiQRQ iqUMFD"
      >
        <div
          className="sc-ksdxAp hXqVet"
        >
          <h5
            className="sc-gKckTs dFXzsw"
          >
            <p
              className="sc-eCImvq eTrCnx"
              color="white"
            >
              株式会社zzz
            </p>
          </h5>
          <h6
            className="sc-iCfLBT ebzyty"
          >
            <p
              className="sc-eCImvq kBjSVF"
              color="#C7C7C7"
            >
              1年間
            </p>
          </h6>
        </div>
        <div
          className="sc-ksdxAp hXqVet"
        >
          <span
            className="sc-hBURRC hTBict"
          >
            <div
              className="sc-furvIG dJoCcd"
            >
              <p
                className="sc-eCImvq dVIdtT"
                color="#424242"
              >
                営業
              </p>
            </div>
          </span>
          <span
            className="sc-hBURRC hTBict"
          >
            <div
              className="sc-furvIG dJoCcd"
            >
              <p
                className="sc-eCImvq dVIdtT"
                color="#424242"
              >
                エンジニア
              </p>
            </div>
          </span>
          <span
            className="sc-hBURRC hTBict"
          >
            <div
              className="sc-furvIG dJoCcd"
            >
              <p
                className="sc-eCImvq dVIdtT"
                color="#424242"
              >
                セールス
              </p>
            </div>
          </span>
          <span
            className="sc-hBURRC hTBict"
          >
            <div
              className="sc-furvIG dJoCcd"
            >
              <p
                className="sc-eCImvq dVIdtT"
                color="#424242"
              >
                労務
              </p>
            </div>
          </span>
        </div>
        <div
          className="sc-ksdxAp hXqVet"
        >
          <h6
            className="sc-iCfLBT ebzyty"
          >
            <p
              className="sc-eCImvq eTrCnx"
              color="white"
            >
              helloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworld
            </p>
          </h6>
        </div>
      </div>
    </div>
  </div>
</div>
`;
